images/hub/jupyterhub_config.py

```diff
@@ -177,6 +178,25 @@ def get_config(key, default=None):
     c.DummyAuthenticator.password = get_config('auth.dummy.password', None)
 elif auth_type == 'tmp':
     c.JupyterHub.authenticator_class = 'tmpauthenticator.TmpAuthenticator'
+elif auth_type == 'custom':
+    # full_class_name looks like "myauthenticator.MyAuthenticator".
+    # To create a docker image with this class availabe, you can just have the
+    # following Dockerifle:
+    #   FROM jupyterhub/k8s-hub:v0.4
+    #   RUN pip3 install myauthenticator
+    full_class_name = get_config('auth.custom.class_name')
+    c.JupyterHub.authenticator_class = full_class_name
+
+    (module, class_name) = full_class_name.split('.')
+    the_module = importlib.import_module(module)
+    klass=getattr(the_module, class_name)
+
+    class_params = get_config('auth.custom.params')
```

Comment:
I like how everything is under `auth` right now. What's under `auth` is simple, but if it was something more complex it might be confusing if part of the parameters are under `auth` and part of them are under `extraConfig`.

Also, I might want to do a quick debugging while developing and switch from custom to dummy.
```bash
auth:
  type: "dummy"   #  this was "custom"
  custom:
    className: "hashauthenticator.HashAuthenticator"
    traits:
      secret_key: "834e6cace4747e966c0.."
```

Now everything under `auth.custom` never reaches Kubernetes, since helm filters it out. If this part of the config was under extraConfig, I wouldn't be able to filter it, I'd have to find it and comment it out under "extraConfig".

Also, everything from extraConfig will get passed verbatim. You might want to have some config parameters that are altered a bit in the code before setting trait values.